sympy/matrices/tests/test_matrices.py

```diff
@@ -2767,6 +2767,25 @@ def test_pinv_rank_deficient():
     assert solution == Matrix([3, w1])
     assert A * A.pinv() * B != B
 
+@XFAIL
+def test_pinv_rank_deficient_when_diagonalization_fails():
+    print('Test the four properties of the pseudoinverse for matrices when diagonalization of A.H*A fails.')
+    As = [Matrix([
+        [0.61, 0.89, 0.55, 0.20, 0.71, 0],
```

Comment:
It would be better if these were sympy numbers or `int`s. As is, they are python floats which get converted to sympy rationals with the exact precision of the float.